Finished part 2

dantevibes opened the pull request:
-Changed all pointer usage to reference types wherever possible.

dantevibes committed:
Finished part 2
-Changed all pointer usage to reference types wherever possible.

matkatmusic commented on main.cpp:
```diff
@@ -33,68 +33,59 @@ T::T(char v, const char* vName) :
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }
-        return nullptr;
+        if( a.value < b.value ) return a;
+        else if( a.value > b.value ) return b;
+        
+        std::cout << "Values are equal, returning first arg..." << std::endl;
+        return a;
```
Follow the spec
```
 1) convert all of the pointer usage (except for 'const char*') to reference types or 
    const reference types **>>> WHERE POSSIBLE <<<**
 hint: There is no reference equivalent to nullptr.  
        if a pointer (including nullptr) is being returned anywhere, 
             don't try to convert it to a reference.
```
- [x] revise

matkatmusic reviewed (changes requested):
Follow the instructions

dantevibes committed:
Changed compare function back to returning a pointer.

matkatmusic commented on main.cpp:
```diff
@@ -38,63 +38,61 @@ struct CompareTValues                                //4
         if(a != nullptr && b != nullptr)
         {
             if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
+            else if( a->value > b->value ) return b;
+            else
+            {
+                std::cout << "Values are equal, returning first arg..." << std::endl;
+                return a;
+            }
         }
+        std::cout << "got a nullptr, returning nullptr" << std::endl;
         return nullptr;
     }
```
This is still not correct.
Ask yourself:
1) can the return type be converted to a reference?  Hint: what do the instructions say?
2) can the function arguments be converted to a reference? 
- [x] revise

matkatmusic reviewed (changes requested):
It's still not correct. 
You need to think carefully about what the instructions are asking you to do with regard to the `compare` function

dantevibes committed:
Changed compare func to
-take references as args
-return a pointer
-return nullptr when values are equal.

dantevibes committed:
Simplified compare function
